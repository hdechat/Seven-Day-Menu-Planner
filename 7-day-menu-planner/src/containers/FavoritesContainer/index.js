import { connect } from 'react-redux';
import { withRouter } from 'react-router';
import Favorites from '../../components/Favorites';

export const mapStateToProps = state => ({
  favorites: state.favorites
});

export default withRouter(connect(mapStateToProps)(Favorites));