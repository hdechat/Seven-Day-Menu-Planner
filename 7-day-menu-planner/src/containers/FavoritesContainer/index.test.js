import React from 'react';
import { shallow } from 'enzyme';
import FavoritesContainer, { mapStateToProps } from './index.js';

describe('FAVORITES CONTAINER', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<FavoritesContainer />);
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should return the prop object', () => {
    const mockState = {
      recipes: {},
      menu: {},
      groceryList: [],
      favorites: [{title: 'recipe'}]
    };

    const expected = {
      favorites: [{title: 'recipe'}]
    };

    const mappedProps = mapStateToProps(mockState);

    expect(mappedProps).toEqual(expected);
  });
});