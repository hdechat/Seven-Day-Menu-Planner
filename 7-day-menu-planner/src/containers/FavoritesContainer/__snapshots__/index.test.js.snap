// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FAVORITES CONTAINER matches snapshot 1`] = `undefined`;
